docs/vector_store.ipynb

```diff
@@ -1,79 +1,356 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rvs = RedisVectorStore(\n",
+    "    client=redis_client, index_name=\"my_vector_index\", embedding_service=embeddings\n",
+    ")\n",
+    "ids = rvs.add_texts(\n",
+    "    texts=[d.page_content for d in docs], metadatas=[d.metadata for d in docs]\n",
+    ")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Perfor a Similarity Search (KNN)\n",
+    "\n",
+    "With the vector store populated, it's possible to search for text semantically similar to a query.  Here's how to use KNN (K-Nearest Neighbors) with default settings:\n",
+    "\n",
+    "* Formulate the Query: A natural language question expresses the search intent (e.g., \"What did the president say about Ketanji Brown Jackson\").\n",
+    "* Retrieve Similar Results: The `similarity_search` method finds items in the vector store closest in meaning to the query."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pprint\n",
+    "\n",
+    "query = \"What did the president say about Ketanji Brown Jackson\"\n",
+    "knn_results = rvs.similarity_search(query=query)\n",
+    "pprint.pprint(knn_results)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Perform a Range-Based Similarity Search\n",
+    "\n",
+    "Range queries provide more control by specifying a desired similarity threshold along with the query text:\n",
+    "\n",
+    "* Formulate the Query: A natural language question defines the search intent.\n",
+    "* Set Similarity Threshold: The distance_threshold parameter determines how close a match needs to be considered relevant.\n",
+    "* Retrieve Results: The `similarity_search_with_score` method finds items from the vector store that fall within the specified similarity threshold."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rq_results = rvs.similarity_search_with_score(query=query, distance_threshold=0.8)\n",
+    "pprint.pprint(rq_results)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Perform a Maximal Marginal Relevance (MMR) Search\n",
+    "\n",
+    "MMR queries aim to find results that are both relevant to the query and diverse from each other, reducing redundancy in search results.\n",
+    "\n",
+    "* Formulate the Query: A natural language question defines the search intent.\n",
+    "* Balance Relevance and Diversity: The lambda_mult parameter controls the trade-off between strict relevance and promoting variety in the results.\n",
```

Comment:
"Balance, Relevance, and Diversity"

Reply:
Balance is a verb in this context and the tradeoff is between "relevance" and "diversity"